src/common.rs

```diff
@@ -99,10 +101,20 @@ where
         let e = self.expected.lock().unwrap();
         assert!(e.is_empty(), "Not all expectations consumed");
     }
+
+    /// Get the mock data
+    pub fn mock_data(&self) -> &Option<Data> {
+        &self.mock_data
+    }
+
+    /// Set the mock data
+    pub fn set_mock_data(&mut self, data: Option<Data>) {
```

Comment:
Would be nice to have some example uses here.

Reply:
Actually, I wasn't sure myself on how to do this properly. Getters and setter feel kinda wrong here, but sharing mut ref can be problematic as well.

Not sure on how to show examples for a getter and setter methods, though.

Added an appropriate comment on top of the module, that describes mock_data functionality.

Reply:
I think those docs should suffice 👍🏻